src/models/ServerStyleSheet.js

```diff
@@ -2,218 +2,104 @@
-  }
-
-  addComponent(componentId: string) {
-    if (this.components[componentId]) {
-      throw new Error(
-        process.env.NODE_ENV !== 'production'
-          ? `Trying to add Component '${componentId}' twice!`
-          : ''
-      )
-    }
-    this.components[componentId] = { componentId, css: '' }
-    this.size += 1
-  }
-
-  concatenateCSS() {
-    return Object.keys(this.components).reduce(
-      (styles, k) => styles + this.components[k].css,
-      ''
-    )
-  }
-
-  inject(componentId: string, css: Array<string>, name: ?string) {
-    const comp = this.components[componentId]
-
-    if (!comp) {
-      throw new Error(
-        process.env.NODE_ENV !== 'production'
-          ? 'Must add a new component before you can inject css into it'
-          : ''
-      )
-    }
-
-    if (comp.css === '') {
-      comp.css = `/* sc-component-id: ${componentId} */\n`
-    }
-
-    const cssRulesSize = css.length
-    for (let i = 0; i < cssRulesSize; i += 1) {
-      const cssRule = css[i]
-      comp.css += `${cssRule}\n`.replace(/\n*$/, '\n')
-    }
-
-    if (name) this.names.push(name)
-  }
-
-  toHTML() {
-    const attrs: Array<string> = [
-      'type="text/css"',
-      `${SC_ATTR}="${this.names.join(' ')}"`,
-      `${LOCAL_ATTR}="${this.isLocal ? 'true' : 'false'}"`,
-    ]
-
-    const nonce = getNonce()
-    if (nonce) {
-      attrs.push(`nonce="${nonce}"`)
-    }
-
-    this.emitted = true
-    return `<style ${attrs.join(' ')}>${this.concatenateCSS()}</style>`
-  }
-
-  toReactElement(key: string) {
-    const attrs: Object = {
-      [SC_ATTR]: this.names.join(' '),
-      [LOCAL_ATTR]: this.isLocal.toString(),
-    }
-
-    const nonce = getNonce()
-    if (nonce) {
-      attrs.nonce = nonce
-    }
-
-    this.emitted = true
-
-    return (
-      <style
-        key={key}
-        type="text/css"
-        {...attrs}
-        dangerouslySetInnerHTML={{ __html: this.concatenateCSS() }}
-      />
-    )
-  }
-
-  clone() {
-    const copy = new ServerTag(this.isLocal)
-    copy.names = [].concat(this.names)
-    copy.size = this.size
-    copy.components = Object.keys(this.components).reduce((acc, key) => {
-      acc[key] = { ...this.components[key] } // eslint-disable-line no-param-reassign
-      return acc
-    }, {})
-
-    return copy
-  }
-}
+/* this error is used for makeStyleTag */
+const sheetClosedErr =
+  process.env.NODE_ENV !== 'production'
+    ? `
+Can't collect styles once you've consumed a ServerStyleSheet's styles!
+ServerStyleSheet is a one off instance for each server-side render cycle.
+- Are you trying to reuse it across renders?
+- Are you accidentally calling collectStyles twice?
+`.trim()
+    : ''
+
+const streamBrowserErr =
+  process.env.NODE_ENV !== 'production'
+    ? 'Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.'
+    : ''
 
 export default class ServerStyleSheet {
   closed: boolean
+  global: StyleSheet
```

Comment:
`s/global/singleton/` ? `global` reminds me too much of `injectGlobal`

Reply:
Or perhaps "master"

Reply:
Typically singletons would indicate that there’s only a single instance... at least that’s what I think of them.

I chose “global” as an opposite to localised Stylesheets since soon we might need to introduce more sophisticated Stylesheet management for the manager’s rehydration.

Totally up to discuss this for sure, but it’d also be no problem to do this in a separate issue/PR :)

Reply:
This still needs to be changed to `master` right?